src/app/stream-deck/actions.ts

```diff
@@ -0,0 +1,366 @@
+    if (!data) {
+      return;
+    }
+
+    const currentStore = currentStoreSelector(state);
+
+    if (!currentStore) {
+      return;
+    }
+
+    switch (data.action) {
+      case 'refresh':
+        return refresh();
+      case 'search': {
+        goToPage(data.args.page || 'inventory');
+        setTimeout(() => dispatch(setSearchQuery(data.args.search, true)), 500);
+        return;
+      }
+      case 'randomize': {
+        const allItems = allItemsSelector(state);
+        const loadout = randomLoadout(
+          currentStore,
+          allItems,
+          data.args.weaponsOnly ? (i) => i.bucket?.sort === 'Weapons' : () => true
+        );
+        loadout && (await dispatch(applyLoadout(currentStore, loadout, { allowUndo: true })));
+        return;
+      }
+      case 'collectPostmaster': {
+        return dispatch(pullFromPostmaster(currentStore));
+      }
+      case 'farmingMode': {
+        if (state.farming.storeId) {
+          return dispatch(stopFarming());
+        } else {
+          return dispatch(startFarming(currentStore?.id));
+        }
+      }
+      case 'maxPower': {
+        const allItems = allItemsSelector(state);
+        const loadout = maxLightLoadout(allItems, currentStore);
+        return dispatch(applyLoadout(currentStore, loadout, { allowUndo: true }));
+      }
+      case 'selection': {
+        const selectionType = data.args.selection;
+        dispatch(setSearchQuery(''));
+        dispatch(streamDeckWaitSelection(selectionType));
+
+        // open the related page
+        goToPage(selectionType === 'loadout' ? 'loadouts' : 'inventory');
+
+        // show the notification
+        setTimeout(
+          () =>
+            showSelectionNotification(state, selectionType, () => {
+              dispatch(streamDeckClearSelection());
+            }),
+          500
+        );
+        return;
+      }
+      case 'loadout': {
+        const loadouts = loadoutsSelector(state);
+        const store = storesSelector(state).find((it) => it.id === data.args.character);
+        const loadout = loadouts.find((it) => it.id === data.args.loadout);
+        if (store && loadout) {
+          return dispatch(applyLoadout(store, loadout, { allowUndo: true }));
+        }
+        return;
+      }
+      case 'shareLoadout': {
+        const loadouts = loadoutsSelector(state);
+        const account = currentAccountSelector(state);
+        const accountId = account?.membershipId;
+        const loadout = loadouts.find((it) => it.id === data.args.loadout);
+        if (accountId && loadout) {
+          const shareUrl =
+            loadoutShares.get(loadout) ||
+            (await createLoadoutShare(accountId, convertDimLoadoutToApiLoadout(loadout)));
+          loadoutShares.set(loadout, shareUrl);
+          return dispatch(
+            sendToStreamDeck({
+              shareUrl,
+            })
+          );
+        }
+        return;
+      }
+      case 'freeSlot': {
```

Comment:
We'd call this a bucket rather than a slot (I always think of a bucket as holding several slots)